app/src/forms/auth/middleware/userAccess.js

```diff
@@ -74,6 +76,60 @@ const hasFormPermissions = (permissions) => {
   };
 };
 
+const hasSubmissionPermissions = (permissions) => {
+  return async (req, res, next) => {
+    if (!Array.isArray(permissions)) {
+      permissions = [permissions];
+    }
+
+    // Get the provided submission ID whether in a param or query (precedence to param)
+    const submissionId = req.params.formSubmissionId || req.query.formSubmissionId;
+    if (!submissionId) {
+      // No submission provided to this route that secures based on form... that's a problem!
+      return next(new Problem(401, { detail: 'Submission Id not found on request.' }));
+    }
+
+    // Get the submission results so we know what form this submission is for
+    const submissionForm = await service.getSubmissionForm(submissionId);
+    const isDeleted = submissionForm.submission.deleted;
+
+    // Deleted submissions are inaccessible
+    if (isDeleted) {
```

Comment:
Doesn't seem like isDeleted is used anywhere else. Could streamline this into
```js
if (submissionForm.submission.deleted) {
```

Reply:
Yeah true